addons/web/static/src/views/kanban/kanban_renderer.js

```diff
@@ -551,13 +551,26 @@ export class KanbanRenderer extends Component {
         // Search current card position
         let iGroup;
         let iCard;
+        // first check that a card has the focus
         for (iGroup = 0; iGroup < cards.length; iGroup++) {
             const i = cards[iGroup].indexOf(document.activeElement);
```

Comment:
What would you think about using `:focus-within` to avoid a second loop ?
```suggestion
        // first check that a card has the focus
        for (iGroup = 0; iGroup < cards.length; iGroup++) {
            const i = cards[iGroup].findIndex(card => card === document.activeElement || card.querySelector(":focus-within"));
```

Reply:
same with contains, it could be done in one shot for less code but degrades the general case. 
btw, I think contains, and :focus-within method should have a similar complexity (if you look at the specs they both use standard tree traversal). The contain version is more readable IMHO.
Now the question here is, do we care about performance and de-optimize the general case ?

Reply:
I see your point about de-optimizing the general case and its potential performance penalty.
Not sure what the impact could be in real world and/or with a huge amount of records displayed...
/invoke @aab-odoo for his opinion

About the readability and the `contains` vs `:focus-within`... I guess my CSS tastes betrays me :nerd_face:

Reply:
What about doing something like `document.activeElement.closest(".o_kanban_card")` ? Then it's just a find index. And btw, the nested `cards` structure is complicated imo, so maybe all these should be flattened, which would even simplify the finding of the next card.

Reply:
I applied your suggestion @aab-odoo, it's the best of performance and readability :)